site/src/components/DurationField/DurationField.tsx

```diff
@@ -0,0 +1,187 @@
+  switch (action.type) {
+    case "SYNC_WITH_PARENT": {
+      return initState(action.parentValueMs);
+    }
+    case "CHANGE_DURATION_FIELD_VALUE": {
+      return {
+        ...state,
+        durationFieldValue: action.fieldValue,
+      };
+    }
+    case "CHANGE_TIME_UNIT": {
+      const currentDurationMs = durationInMs(
+        state.durationFieldValue,
+        state.unit,
+      );
+
+      if (
+        action.unit === "days" &&
+        !canConvertDurationToDays(currentDurationMs)
+      ) {
+        return state;
+      }
+
+      return {
+        unit: action.unit,
+        durationFieldValue:
+          action.unit === "hours"
+            ? durationInHours(currentDurationMs).toString()
+            : durationInDays(currentDurationMs).toString(),
+      };
+    }
+    default: {
+      return state;
+    }
+  }
+};
+
+export const DurationField: FC<DurationFieldProps> = (props) => {
+  const {
+    valueMs: parentValueMs,
+    onChange,
+    helperText,
+    ...textFieldProps
+  } = props;
+  const [state, dispatch] = useReducer(reducer, initState(parentValueMs));
+  const currentDurationMs = durationInMs(state.durationFieldValue, state.unit);
+
+  useEffect(() => {
+    if (parentValueMs !== currentDurationMs) {
+      dispatch({ type: "SYNC_WITH_PARENT", parentValueMs });
+    }
+  }, [currentDurationMs, parentValueMs]);
+
+  return (
+    <div>
+      <div
+        css={{
+          display: "flex",
+          gap: 8,
+        }}
+      >
+        <TextField
+          {...textFieldProps}
+          fullWidth
+          value={state.durationFieldValue}
+          onChange={(e) => {
+            const durationFieldValue = intMask(e.currentTarget.value);
+
+            dispatch({
+              type: "CHANGE_DURATION_FIELD_VALUE",
+              fieldValue: durationFieldValue,
+            });
+
+            const newDurationInMs = durationInMs(
+              durationFieldValue,
+              state.unit,
+            );
+            if (newDurationInMs !== parentValueMs) {
+              onChange(newDurationInMs);
+            }
+          }}
+          inputProps={{
+            step: 1,
+          }}
+        />
+        <Select
+          disabled={props.disabled}
+          css={{ width: 120, "& .MuiSelect-icon": { padding: 2 } }}
+          value={state.unit}
+          onChange={(e) => {
+            const unit = e.target.value as TimeUnit;
+            dispatch({
+              type: "CHANGE_TIME_UNIT",
+              unit,
+            });
+          }}
+          inputProps={{ "aria-label": "Time unit" }}
+          IconComponent={KeyboardArrowDown}
+        >
+          <MenuItem value="hours">Hours</MenuItem>
+          <MenuItem
+            value="days"
+            disabled={!canConvertDurationToDays(currentDurationMs)}
+          >
+            Days
+          </MenuItem>
+        </Select>
+      </div>
+
+      {helperText && (
+        <FormHelperText error={props.error}>{helperText}</FormHelperText>
```

Comment:
Was there a reason why `props.error` wasn't destructured from the props at the top of the component?

Reply:
Good catch!